game.py

```diff
@@ -0,0 +1,291 @@
+class SnakeGame(GameManager):
+    def __init__(self,**kwargs):
+        super(SnakeGame,self).__init__(**kwargs)
+
+    def is_game_over(self,snake):
+        if snake.x >= self.display_width or snake.x < 0 or snake.y >= self.display_height or snake.y < 0:
+            return True
+        return False
+
+    def fill_blue(self):
+        self.game_display.fill(self.sky_blue)
+
+    # Initliazes text object
+    def text_objects(self, text, color):
+        textSurface = self.font.render(text, True, color)
+        return textSurface, textSurface.get_rect()
+
+    # Defines  display message with color variable
+    def message_to_screen(self, msg, color, y_displace=0):
+        textSurf, textRect = self.text_objects(msg,color)
+        textRect.center = (self.display_width / 2), (self.display_height / 2) + y_displace
+        self.game_display.blit(textSurf, textRect)
+    #places score on bottom right corner
+    def score_to_screen(self, msg, color):
+        textSurf, textRect = self.text_objects(msg, color)
+        textRect = (self.display_width - 50), (self.display_height - 50)
+        self.game_display.blit(textSurf, textRect)
+
+class Image(object):
+    def __init__(self,**kwargs):
+        self.start_pos = kwargs.pop('start_pos') #(x,y)
+        self.icon = pygame.image.load(kwargs.pop('icon'))
+        self.icon = pygame.transform.scale(self.icon, (kwargs.pop('image_width'), kwargs.pop('image_height')))
+        self.x = self.start_pos[0]
+        self.y = self.start_pos[1]
+        if 'image_width' in kwargs:  
+            self.image_width = kwargs.pop('image_width')
+        else:
+            self.image_width = 50 
+        if 'image_height' in kwargs:  
+            self.image_height = kwargs.pop('image_height')
+        else:
+            self.image_height = 50
+
+
+#start_pos=(0,0),icon='food.png', image_width=50, image_height=50
+
+#init "food" image
+class Carrot(Image):
+    def __init__(self,**kwargs):
+        super(Carrot,self).__init__(**kwargs)
+    
+    def set_random_location(self,display_width,display_height):
+        self.x = round(random.randrange(0, display_width-self.image_width))
+        self.y = round(random.randrange(0, display_height-self.image_height))
+        self.start_pos = (self.x,self.y)
+#init rainbow image
+class Rainbow(Image):
+    def __init__(self, **kwargs):
+        super(Rainbow,self).__init__(**kwargs)
+
+#init unicorn/snake image
+class Snake(Image):
+    def __init__(self,**kwargs):
+        super(Snake,self).__init__(**kwargs)
+        self.tail_list = []
+        self.tail_size = 1
+        self.direction = "right"
+        self.delta_y = 0
+        self.delta_x = 40
+	
+    #function to add rainbow to tail
+    def add_to_tail(self):
+        gameOver = False
+        snake_head = []
+        snake_head.append(self.x)
+        snake_head.append(self.y)
+
+        self.tail_list.append(snake_head)
+
+        if len(self.tail_list) > self.tail_size:
+            del self.tail_list[0]
```

Comment:
It's often "safer" to use list methods like .pop(0), rather than python's built-in `del` keyword. If you're working with massive datasets, removing the first element of a list is also not ideal due to the way that python handles lists -- it's not relevant here, since tails won't wind up being hundreds or thousands of items long, but if you're interested in other ways of storing/handling data efficiently I recommend reading about double-ended queues, or `deque`s, e.g. [here](https://docs.python.org/2/library/collections.html).